pkg/engine/provider/extract.go

```diff
@@ -0,0 +1,185 @@
+package provider
+
+import (
+	"context"
+	"io/fs"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"sync"
+
+	"github.com/Checkmarx/kics/pkg/model"
+	"github.com/rs/zerolog/log"
+
+	"github.com/hashicorp/go-getter"
+)
+
+const (
+	channelLength = 2
+)
+
+// ExtractedPath is a struct that contains the paths, temporary paths to remove
+// and extraction map path of the sources
+// Path is the slice of paths to scan
+// ExtrectionMap is a map that correlates the temporary path to the givven path
+// RemoveTmp is the slice containing temporary paths to be removed
+type ExtractedPath struct {
+	Path          []string
+	ExtrectionMap map[string]model.ExtractedPathObject
+	RemoveTmp     []string
+}
+
+type getterStruct struct {
+	ctx         context.Context
+	cancel      context.CancelFunc
+	mode        getter.ClientMode
+	pwd         string
+	opts        []getter.ClientOption
+	destination string
+	source      string
+}
+
+// GetSources goes through the source slice, and determines the of source type (ex: zip, git, local).
+// It than extracts the files to be scanned. If the source given is not local, a temp dir
+// will be created where the files will be stored.
+func GetSources(source []string) (ExtractedPath, error) {
+	returnStr := ExtractedPath{
+		Path:          []string{},
+		ExtrectionMap: make(map[string]model.ExtractedPathObject),
+		RemoveTmp:     []string{},
+	}
+	for _, path := range source {
+		destination, err := os.MkdirTemp("", "kics-extract-*")
+		if err != nil {
+			return ExtractedPath{}, err
+		}
+
+		err = os.RemoveAll(destination)
```

Comment:
I think we could change this to guarantee a unique name, like a timestamp, for example, instead of trying to remove if it exists...